src/pymovements/gaze/experiment.py

```diff
@@ -109,12 +115,17 @@ def __init__(
             origin=origin,
         )
 
-        checks.check_is_not_none(sampling_rate=sampling_rate)
-        assert sampling_rate is not None
+        checks.check_is_mutual_exclusive(sampling_rate=sampling_rate, eyetracker=eyetracker)
+
+        if sampling_rate is not None:
```

Comment:
I reconsidered a bit how we should implement this short-access. I worry that this is a bit too unsafe, because people could do something strange like updating the sampling rate which would introduce an inconsistency:
```python
experiment = Experiment(..., experiment=EyeTracker(1000))
experiment.eyetracker.sampling_rate = 12345.0
print(experiment.sampling_rate)
1000
```

I think it would be best to replace both if-clauses just with this:
```python
self._sampling_rate = sampling_rate
```
and then create a property for the sampling_rate that chooses the correct one:
```python
@property
def sampling_rate(self): 
    if self._sampling_rate is not None:
        return self._sampling_rate

    return self.eyetracker.sampling_rate
```

By storing the explicitly passed `sampling_rate` in a private attribute we can infer the value dynamically.

If you want, you can also include a `@sampling_rate.setter` method to set the sampling rate, but on assignment we should update `eyetracker.sampling_rate` too.
It's also ok to have this attribute read-only. Do as you wish on that case.

Reply:
I just noticed that implementing it this way will remove the sampling rate from the `__str__` string generated by the `@auto_str` decorator. So when working on this, you need to implement a short `__str__` method of the `Experiment` class. 

https://github.com/aeye-lab/pymovements/blob/f9c43ddfbd0212edfd409e8a3ad94844af5023ec/src/pymovements/utils/decorators.py#L36-L39

I'd propose this a simple implementation:

```python
    def __str__(self: Any) -> str:
        attributes = ', '.join(f'{key}={shorten(value)}' for key, value in vars(self).items())
        return f'{type(self).__name__}(sampling_rate={self.sampling_rate}, {attributes})'
```

The `@auto_str` decorator must be removed from the class then.